os/arch/arm/src/tiva/tiva_flash.c

```diff
@@ -197,12 +208,23 @@ static ssize_t tiva_bread(FAR struct mtd_dev_s *dev, off_t startblock, size_t nb
 
 static ssize_t tiva_bwrite(FAR struct mtd_dev_s *dev, off_t startblock, size_t nblocks, FAR const uint8_t *buf)
 {
+
+#if !defined(CONFIG_QEMU_SRAM) &&  !defined (CONFIG_QEMU_SDRAM)
 	FAR uint32_t *src = (uint32_t *) buf;
 	FAR uint32_t *dst = (uint32_t *)(TIVA_VIRTUAL_BASE + startblock * TIVA_FLASH_PAGESIZE);
 	int i;
+#endif
 
 	DEBUGASSERT(nblocks <= TIVA_VIRTUAL_NPAGES);
 
+#if defined(CONFIG_QEMU_SRAM) || defined (CONFIG_QEMU_SDRAM)
+	/* Qemu doesnt implement tive flash memory controller.
+	 * Hence copy the contents from buf to flash blocks using memcpy.
+	 */
+	memcpy((uint32_t *)(TIVA_VIRTUAL_BASE + startblock * TIVA_FLASH_PAGESIZE), (uint32_t *)buf, nblocks * TIVA_FLASH_PAGESIZE);
```

Comment:
You could have used src and dest variables declared above . Is there any reason for binding them under not defined flag and not using it inside memcpy ?  May be am wrong. just curious to know